syntax = "proto3";

package Proto;

message Vector3{
    float x = 1;
    float y = 2;
    float z = 3;
}

enum Gender{
    none = 0;
    male = 1;
    female = 2;
}

message PlayerBase{
    Gender gender = 1;
    int32 level = 2;
}

message LastWorld{
    int32 worl_id = 1;
    int64 world_sn = 2;
    Vector3 position = 3;
}

message PlayerMisc{
    LastWorld last_world = 1;
    LastWorld last_dungeon = 2;
    int32 online_version = 3;
}

message Player{
    uint64 sn = 1;
    string name = 2;
    PlayerBase base = 3;
    PlayerMisc misc = 4;
}

